File: app/src/main/java/com/aaronbrecher/neverlate/viewmodels/DetailActivityViewModel.java
package com.aaronbrecher.neverlate.viewmodels;

import androidx.lifecycle.MutableLiveData;

import com.aaronbrecher.neverlate.AppExecutors;
import com.aaronbrecher.neverlate.database.EventsRepository;
import com.aaronbrecher.neverlate.AwarenessFencesCreator;
import com.aaronbrecher.neverlate.models.Event;

import java.util.ArrayList;

import javax.inject.Inject;

public class DetailActivityViewModel extends BaseViewModel {
    private MutableLiveData<Event> mEvent;

    @Inject
    public DetailActivityViewModel(EventsRepository eventsRepository, AppExecutors appExecutors) {
        super(eventsRepository, null, appExecutors);
    }

    public MutableLiveData<Event> getEvent() {
        if (mEvent == null)
            mEvent = new MutableLiveData<>();
        return mEvent;
    }

    public void setEvent(Event event) {
        if (mEvent == null) {
            mEvent = new MutableLiveData<>();
        }
        mEvent.postValue(event);
    }

    public void updateEvent(Event event) {
        mAppExecutors.diskIO().execute(() -> mEventsRepository.updateEvents(event));
    }

    public void resetFenceForEvent(Event event){
        ArrayList<Event> events = new ArrayList<>();
        events.add(event);
        AwarenessFencesCreator creator = new AwarenessFencesCreator.Builder(events).build();
        creator.buildAndSaveFences();
    }

    public void removeGeofenceForEvent(Event event){
        AwarenessFencesCreator creator = new AwarenessFencesCreator.Builder(null).build();
        creator.removeFences(event);
    }

}

File: app/src/main/java/com/aaronbrecher/neverlate/viewmodels/DetailActivityViewModel.kt
package com.aaronbrecher.neverlate.viewmodels

import com.aaronbrecher.neverlate.AppExecutors
import com.aaronbrecher.neverlate.AwarenessFencesCreator
import com.aaronbrecher.neverlate.database.EventsRepository
import com.aaronbrecher.neverlate.models.Event
import java.util.*
import javax.inject.Inject

class DetailActivityViewModel @Inject
constructor(eventsRepository: EventsRepository, appExecutors: AppExecutors) : BaseViewModel(eventsRepository, null, appExecutors) {

    fun updateEvent(event: Event) {
        mAppExecutors.diskIO().execute { mEventsRepository.updateEvents(event) }
    }

    fun resetFenceForEventForTransportChange(event: Event) {
        val events = listOf(event)
        val creator = AwarenessFencesCreator.Builder(events).build()
        creator.buildAndSaveFences(true)
    }

    fun removeGeofenceForEvent(event: Event) {
        val creator = AwarenessFencesCreator.Builder(null).build()
        creator.removeFences(event)
    }

}
